mlir/utils/jenkins/MIOpenDriver.py

```diff
@@ -11,29 +12,91 @@
 import math
 import itertools
 from datetime import date
+from pathlib import Path
+import glob
+import argparse
 
 import numpy as np
 import pandas as pd
 import re
 
 # global variables.
-MLIR_BIN_DIR = './bin'
-LLVM_BIN_DIR = './external/llvm-project/llvm/bin'
-MIOPEN_GEN = 'miopen-gen'
-MLIR_MIOPEN_DRIVER = 'mlir-miopen-driver'
-MLIR_ROCM_RUNNER = 'mlir-rocm-runner'
-MLIR_ROCM_RUNNER_ARGS = ['--shared-libs=./external/llvm-project/llvm/lib/libmlir_rocm_runtime.so,./lib/libconv-validation-wrappers.so,./external/llvm-project/llvm/lib/libmlir_runner_utils.so', '--entry-point-result=void']
 ROCPROF = '/opt/rocm/bin/rocprof'
-MIOPEN_DRIVER = '../MIOpen/build/bin/MIOpenDriver'
 BENCHMARKING_RESULT_FILE_NAME = 'results.stats.csv'
-CONFIGURATION_FILE_NAME ='../mlir/utils/jenkins/miopen-tests/resnet50-miopen-configs'
-
 DIRECTIONS = ['-F 1', '-F 2', '-F 4']
 DATA_TYPES = ['conv', 'convfp16', 'convint8']
 LAYOUTS = ['NHWC', 'NCHW']
+
 # Compiled regexp object used for extracting elapsed time from MIOpenDriver's output
 ELAPSED_TIME_RE = re.compile(r"Elapsed: (.*)ms")
 
+class Paths(NamedTuple):
+    """This structure is used to hold paths needed to perform the tests"""
+    miopen_gen_path : str
+    mlir_miopen_driver_path : str
+    miopen_driver_path : str
+    rocm_runner_path : str
+    libmlir_rocm_runtime_path : str
+    libconv_validation_wrappers_path : str
+    libmlir_runtime_utils_path : str
+    configuration_file_path : str
+
+
+def find_mlir_build_dir() -> str:
+    """
+    Finds mlir build dir searching either WORKSPACE dir
+    or home dir
+    """
+    workspace_dir = os.environ.get('WORKSPACE', str(Path.home()))
+    assert workspace_dir, "Cant find WORKSPACE env arg or home directory"
+    miopen_opt_path = glob.glob(workspace_dir + '/**/bin/miopen-opt', recursive=True)
+    assert len(miopen_opt_path) == 1, "Multiple paths found to contain */bin/miopen-opt"
+    miopen_opt_path = miopen_opt_path[0]
+    build_dir = Path(miopen_opt_path).parent.parent
+    return str(build_dir)
+
+
+def find_miopen_build_dir() -> str:
```

Comment:
I'd start this by looking relative to `mlir_build_dir` first

Reply:
any reason ?
It could be anywhere right ?

Reply:
While it could be anywhere, I'd look just under the root of the MLIR repository first. That's where the CI puts it and where a lot of people have it symlinked in, so that'll save on glob calls

Reply:
Makes sense! I ll change this tommorow when Im bored trying to figuring out other things.

Reply:
Done (I think)